src/formatDate.js

```diff
@@ -7,8 +7,70 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const newDateArr = [];
+  let month = '';
+  let day = '';
+  let year = '';
+
+  const separator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateArr = date.split(separator);
+
+  const requiredYearFormat = toFormat.find(
+    (element) => element === 'YYYY' || element === 'YY',
+  );
+
+  const changeYearFormat = () => {
+    if (requiredYearFormat.length === 4 && year.length === 4) {
+      return year;
+    }
+
+    if (requiredYearFormat.length === 4 && year < 30) {
+      return `20${year}`;
+    }
+
+    if (requiredYearFormat.length === 4 && year >= 30) {
+      return `19${year}`;
+    }
+
+    if (requiredYearFormat.length === 2) {
+      return year.slice(-2);
+    }
+  };
+
+  for (let i = 0; i < fromFormat.length; i++) {
```

Comment:
The for-loop used to extract date parts based on 'fromFormat' is correct, but consider using the 'reduce' method or another higher-order function to map 'fromFormat' to the date parts. This can make the code more declarative and easier to understand.